lib/Doctrine/DBAL/Cache/ArrayStatement.php

```diff
@@ -19,7 +20,7 @@
 /**
  * @deprecated
  */
-class ArrayStatement implements IteratorAggregate, ResultStatement, Result
+class ArrayStatement implements IteratorAggregate, ResultStatement, V3Result
```

Comment:
This is still a breaking change.

Reply:
Sorry, I don't understand why. This class still implements Result through V3Result. By the way, I used roave_bc_check tool and it did not throw BC break warning anymore. Can you explain it to me please ? 

Thanks again for your quick feedback.

Reply:
Sorry, you're right. I didn't notice that `V3Result` extends `Result`. I'll try to review the changes more thoroughly in the next few days. In the meanwhile, please fix the Psalm failure and provide a couple of examples of the code that gets analyzed incorrectly w/o these changes.